guides/hyperlane-use-tia-for-gas.md

```diff
@@ -0,0 +1,446 @@
+
+```bash
+wasmd keys add my-key
+```
+
+To get your Stride address, import the generated seed phrase to the `strided` CLI using:
+
+```bash
+strided keys add --recover my-key
+```
+
+Then fund it using the [Stride testnet faucet](https://stride-faucet.pages.dev).
+
+To get the private key, run:
+
+```bash
+wasmd keys export --unsafe --unarmored-hex my-key
+```
+
+:::
+
+#### Config the `wasmd` CLI for ease of use:
+
+```bash
+wasmd config set client node tcp://127.0.0.1:36657
+wasmd config set client output json
+wasmd config set client keyring-backend test
+```
+
+#### Fund the cw-hyperlane signer in our localwasmd rollup:
+
+```bash
+wasmd tx bank send localwasm-key wasm133xh839fjn9wxzg6vhc0370lcem8939zr8uu45 10000000uwasm -y --gas auto --gas-adjustment 1.2 --gas-prices 0.025uwasm
+```
+
+:::info
+`localwasm-key` is the genesis validator's address. See https://rollkit.dev/cosmwasm/init.sh for more info.
+:::
+
+:::tip
+To check if the transaction was successful, get the `txhash` field from the response to the `wasmd tx bank send` command above, and run:
+
+```bash
+wasmd q tx $TXHASH
+```
+
+The transaction was successful if the `code` field is 0 (success).
+:::
+
+#### Inside the cw-hyperlane directory, build the Hyperlane contracts:
+
+```bash
+yarn install
+
+# Build contracts from local environment
+make optimize
+# Run compatibility test
+make check
+```
+
+#### Upload and deploy the contracts on our localwasmd rollup:
+
+```bash
+# This command will make one file.
+# - context with artifacts (default path: {cw-hyperlane-root}/context/localwasmd.json)
+yarn cw-hpl upload local -n localwasmd
+
+# This command will output two results.
+# - context + deployment    (default path: {cw-hyperlane-root}/context/localwasmd.json)
+# - Hyperlane agent-config  (default path: {cw-hyperlane-root}/context/localwasmd.config.json)
+yarn cw-hpl deploy -n localwasmd
+```
+
+#### Upload and deploy the contracts on the Stride testnet:
+
+```bash
+# Stride has permissioned CosmWasm, meaning only certain addresses can upload contracts.
+# The Hyperlane contracts have already been uploaded, so all that's left is to instantiate them.
+# This command will initialize the config for stride-testnet-1 with the code IDs of the Hyperlane contracts.
+echo '{
+  "artifacts": {
+    "hpl_mailbox": 362,
+    "hpl_validator_announce": 363,
+    "hpl_ism_aggregate": 364,
+    "hpl_ism_multisig": 365,
+    "hpl_ism_pausable": 366,
+    "hpl_ism_routing": 367,
+    "hpl_igp": 368,
+    "hpl_hook_aggregate": 369,
+    "hpl_hook_fee": 370,
+    "hpl_hook_merkle": 371,
+    "hpl_hook_pausable": 372,
+    "hpl_hook_routing": 373,
+    "hpl_hook_routing_custom": 374,
+    "hpl_hook_routing_fallback": 375,
+    "hpl_test_mock_hook": 376,
+    "hpl_test_mock_ism": 377,
+    "hpl_test_mock_msg_receiver": 378,
+    "hpl_igp_oracle": 379,
+    "hpl_warp_cw20": 380,
+    "hpl_warp_native": 381
+  },
+  "deployments": {}
+}' > context/stride-internal-1.json
```

Comment:
friendly reminder to use the context that we already have (once you're done testing)

The only thing that should be different is the ISM address will be empty and the warp contracts will be empty. Then the deploy step below should naturally only deploy the ISM